ldms/src/ldmsd/ldmsd_stream.c

```diff
@@ -281,7 +281,13 @@ int ldmsd_stream_publish(ldms_t xprt,
 
 	rc = stream_send(xprt, buf, msg_no, 0, (char *)data, data_len);
```

Comment:
I think in the case of `data[data_len-1] != '\0'`, the `data_len` here should be `data_len+1`.